T18941 shows a flat list of test plan runs

alexandrasp opened the pull request:
Fix the view that shows a flat list of test plan runs.

Commit:
flat list of test plans by board/kernel

created a flat-list that show plans
related to each combination of board
and kernel, link it to its own test plan
and remove dead code.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker reviewed (changes requested):
It's on staging: https://staging.kernelci.org/test/

First issues I could see:
* the "arch" column is there twice
* the "plan" column is missing
* the "defconfig" column seems to be the short one, not defconfig_full
* column values are center-aligned, it would make it easier to read left-aligned like on the "boots" view
* the "status" column is missing
* the page title is still "Available Test Group Reports" - it should now be "Available Test Results"
* the "date" column is missing
* the first row is for some old results `staging-20200220.4`, not the latest one so it looks like sorting by date is not working

gctucker commented:
@alexandrasp I've copied the list of issues from my previous comment to a check-list in the description, so they can be ticked as they get addressed.

I haven't really reviewed the code in detail yet, only looked at the web page as a first review.

gctucker commented:
@alexandrasp The latest test results still don't appear in the table, it looks like the date range is still broken.

@alexandrasp What is the "storage test status" item about on the check list?

alexandrasp commented:
> @alexandrasp The latest test results still don't appear in the table, it looks like the date range is still broken.
Ok, I will check.

> 
> @alexandrasp What is the "storage test status" item about on the check list?

It is about save status to have this information on table when you go to the next test result page, or change the order.

alexandrasp commented:
@gctucker, you mention on list that links from each row in the table aren't working...should we have any other link? Plan details link is working on staging.

alexandrasp commented:
@gctucker, you mention on list that links from each row in the table aren't working...should it have any other link? Plan details link is working on staging.

alexandrasp commented:
> @gctucker, you mention on list that links from each row in the table aren't working...should it have any other link? Plan details link is working on staging.

links are done!

alexandrasp commented:
> @gctucker, you mention on list that links from each row in the table aren't working...should it have any other link? Plan details link is working on staging.


links are ready!

gctucker commented:
The data in the table is still not correct.

There is still only a per-row link, we should have links for items in the table like on the builds view.

alexandrasp commented:
> The data in the table is still not correct.
> 
> There is still only a per-row link, we should have links for items in the table like on the builds view.

@gctucker try a hard refresh, maybe clean your cache. Here I can see link in tree and kernel columns.  Only problem is that the data is not in the right order yet..but today I will push something new to see what will happen on staging.

gctucker commented:
> > The data in the table is still not correct.
> > There is still only a per-row link, we should have links for items in the table like on the builds view.
> 
> @gctucker try a hard refresh, maybe clean your cache. Here I can see link in tree and kernel columns. Only problem is that the data is not in the right order yet..but today I will push something new to see what will happen on staging.

OK I can see links in these 2 columns.  But I think the row link should be disabled, to only use the "magnifying class" icon as a link to the detailed results and individual items in the row for other links.

alexandrasp commented:
> > > The data in the table is still not correct.
> > > There is still only a per-row link, we should have links for items in the table like on the builds view.
> > 
> > 
> > @gctucker try a hard refresh, maybe clean your cache. Here I can see link in tree and kernel columns. Only problem is that the data is not in the right order yet..but today I will push something new to see what will happen on staging.
> 
> OK I can see links in these 2 columns. But I think the row link should be disabled, to only use the "magnifying class" icon as a link to the detailed results and individual items in the row for other links.

I see

alexandrasp commented:
I think it is done. Solving order by date was really easier than I thought and was trying. @gctucker when you are available take a look on links and review it all.

Commit:
add link in tree, branch and kernel columns

it adds links to /job/selected_job, /job/selected_job/
branch/selected_branch and /build/selected_build/branch/
selected_branch  on table of test-plan runs.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
defining /test/... routes

Add routes to /test/ subdomains:
/test/job/, /test/job/branch/,
/test/job/branch/kernel/ and
/test/job/branch/kernel/plan.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker commented:
Thanks, I think the query is good now.  The data seems complete and in chronological order, the search box works as expected and the table can sort results.  A couple of issues still:

The tree and branch links are not correct, they go to these URLs:
https://staging.kernelci.org/test/job/khilman/
https://staging.kernelci.org/test/job/khilman/branch/to-build/
instead of:
https://staging.kernelci.org/job/khilman/
https://staging.kernelci.org/job/khilman/branch/to-build/

The status is not correct, for example this "sleep" test plan had some failures but the status is shown as "good" in the /test/ view:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200406.1/plan/sleep/

The query to determine a test plan status is to look for regressions for the kernel revision and test plan name, like on this kind of view:
https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200406.1/

alexandrasp commented:
> Thanks, I think the query is good now. The data seems complete and in chronological order, the search box works as expected and the table can sort results. A couple of issues still:
> 
> The tree and branch links are not correct, they go to these URLs:
> https://staging.kernelci.org/test/job/khilman/
> https://staging.kernelci.org/test/job/khilman/branch/to-build/
> instead of:
> https://staging.kernelci.org/job/khilman/
> https://staging.kernelci.org/job/khilman/branch/to-build/
> 
> The status is not correct, for example this "sleep" test plan had some failures but the status is shown as "good" in the /test/ view:
> https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200406.1/plan/sleep/
> 
> The query to determine a test plan status is to look for regressions for the kernel revision and test plan name, like on this kind of view:
> https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200406.1/

Updated. Let's check again on next build on staging.

alexandrasp commented:
@gctucker tested on staging.

Results, status and links seem to be right.

Baseline fails on  kernel staging-20200408.0 and it appears on list as status fail.

https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200408.0/

https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200408.0/plan/baseline/

gctucker commented:
Great, thanks.  I think there are a couple of minor things left to tweak, to remove the "clickable" class on the table and rows are sorted by date but not by time.

gctucker commented on app/dashboard/static/js/app/view-tests-all.2018.9.js:
```diff
@@ -283,45 +163,56 @@ require([
-                .order([0, 'asc'])
+                .order([4, 'desc'])
                 .languageLengthMenu('Tests per page')
-                .rowURL('/test/board/%(board)s/')
```
Actually, maybe this URL could be the same as in the "plan" column to go to a URL like `https://staging.kernelci.org/test/job/kernelci/branch/staging.kernelci.org/kernel/staging-20200408.1/plan/baseline`
so `/test/job/%(job)s/kernel/%(kernel)s/plan/$(plan)s/`

alexandrasp replied:
sure, do you mean : /test/job/%(job)s/branch/%(branch)s/kernel/%(kernel)s/plan/%(plan)s/ or a new view? I updated to go to the link that we already have.

gctucker replied:
Yes, with `/branch/` in the middle... sorry these URLs are rather long.

gctucker commented on app/dashboard/static/js/app/view-tests-all.2018.9.js:
```diff
@@ -283,45 +163,56 @@ require([
                 .data(response)
                 .columns(columns)
-                .order([0, 'asc'])
+                .order([4, 'desc'])
```
There appears to be an issue as rows are sorted by date but not by time.  So branches that have more than one revision in one day will appear in random order.  This is typically the case with the staging branch since there are 3 revisions being tested every day.

The boots view doesn't show this issue: https://staging.kernelci.org/boot/
You can see all the `staging-20200408.1` results at the top, and the `staging-20200408.0` results a few pages down.  Maybe this is because we're aggregating results, whereas the boots view doesn't - I'm not sure.  It's not a big issue but it would be a lot better to be able to sort this in true chronological order.

I think the way this is supposed to work, is that the `gTablesUtils.renderDate()` method will either return the string version if the table is asking for a `display` type of rendering or the full ISO timestamp otherwise.  The `display` type should be used to show a string in the table, and the plain type should be used when ordering the rows.  Maybe that's something to debug and check it's doing the correct thing.

alexandrasp replied:
I am hacking on this :)

gctucker replied:
OK that seems to be fixed now, thanks.

gctucker reviewed (commented):
Some fine-tuning, this looks fine otherwise.

The status for each row does seem to be taking a long time to load, but that may just be because we need another index in the Mongo DB so I'll take a look into this.  It's not a blocker.

Commit:
add link on row

add a link on each table row to
view: /test/branch/kernel/plan.
There you can see details about that
plan considering board, defconfig
arch and compiler infos.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

Commit:
created on order by datetime

it orders created on considering date time,
so the data will be shown on table in
chronological order.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

alexandrasp commented on app/dashboard/static/js/app/tables/common.js:
```diff
@@ -276,7 +276,7 @@ define([
             } else {
                 created = new Date(date.$date);
-                rendered = created.toCustomISODate();
+                rendered = created.toCustomISODateTime();
```
@gctucker I updated this method to consider date time, local it was fine..I also looked for side-effects in builds and boot views and seems to be ok! Let's check on staging, because it's used in many places.

gctucker replied:
But, the other tables didn't have an issue.  So why fix the common.js file which is used by other tables?

alexandrasp replied:
because all tables are using `common.js`, most part of the methods from boot, build, etc. If I am not wrong, every time `renderDate` is called actually it is calling `renderDate` in `common.js` file and now all files using common are calling `.toCustomISODateTime()`.

gctucker replied:
Yes, but before you made this change, the other tables were working fine.  So I don't think this needs to be changed.

alexandrasp replied:
ops, yes! You are right.

gctucker commented:
@alexandrasp I've updated the list of things to fix before this gets merged.  It's pretty close now :)

Commit:
add tooltips

add tooltips for the tree and branch
columns.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

gctucker commented on app/dashboard/static/js/app/view-tests-all.2018.9.js:
```diff
@@ -28,251 +31,134 @@ require([
-                    rendered = gGroupsCount['groups-count-' + data];
+                tooltipNode = html.tooltip();
+                tooltipNode.setAttribute('title', data);
+                if (href && href.length > 0) {
```
There is no `href` in this case, it looks like this was pasted from somewhere else.  At the moment, these links are not being created.  The `_renderBranch()` method should be very similar to the `_renderKernel()` one, with just the branch name added to the URL.

alexandrasp replied:
@gctucker I updated `_renderBranch` to use `renderKernel` and avoid to duplicate code in cases like that, is it ok? I think we can update the code soon to make this kind of function more "generic", so it will be ok use in those cases.

gctucker commented on app/dashboard/static/js/app/view-tests-all.2018.9.js:
```diff
@@ -28,251 +31,134 @@ require([
-        // Internal wrapper to provide the oreder count.
-        function _renderGroupsCount(data, type) {
-            var rendered;
+        function _renderKernel(data, type, object) {
```
We should be doing the same thing as in this view, using the job table `renderKernel()` method:
https://github.com/kernelci/kernelci-frontend/blob/master/app/dashboard/static/js/app/view-jobs-job.2017.4.js#L378

alexandrasp replied:
updated

gctucker commented on app/dashboard/static/js/app/view-tests-all.2018.9.js:
```diff
@@ -28,251 +31,134 @@ require([
-    function getTestsDone(response) {
+    function updateTestTable(response) {
         var columns;
+        function _renderPlan(data, type, object) {
```
Using a `renderKernel()` method to create a link to test plan results seems wrong.  Also, this is actually the same link as the default row link so I think we can just drop it.

alexandrasp replied:
updated.

gctucker reviewed (approved):
Tested OK on staging, LGTM (apart from some whitespace noise but that's OK).
